src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_controller.js

```diff
@@ -0,0 +1,63 @@
+const aggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+
+const parentPipelineAggController = function ($scope) {
+
+  $scope.safeMakeLabel = function (agg) {
+    try {
+      return agg.makeLabel();
+    } catch (e) {
+      return '- agg not valid -';
+    }
+  };
+
+  $scope.$watch('responseValueAggs', updateOrderAgg);
+  $scope.$watch('agg.params.metricAgg', updateOrderAgg);
+
+  $scope.$on('$destroy', function () {
+    if ($scope.aggForm && $scope.aggForm.agg) {
+      $scope.aggForm.agg.$setValidity('bucket', true);
+    }
+  });
+
+  // Returns true if the agg is not compatible with the terms bucket
+  $scope.rejectAgg = function (agg) {
+    // aggFilter elements all starts with a '!'
+    // so the index of agg.type.name in a filter is 1 if it is included
+    return Boolean(aggFilter.find((filter) => filter.indexOf(agg.type.name) === 1));
+  };
+
+  function checkBuckets() {
+    const buckets = $scope.vis.aggs.filter(agg => agg.schema.group === 'buckets');
+    const bucketHasType = buckets.length && buckets[0].type;
+    const bucketIsHistogram = bucketHasType && ['date_histogram', 'histogram'].includes(buckets[0].type.name);
+    const canUseAggregation = buckets.length === 1 && bucketIsHistogram;
+    if ($scope.aggForm.agg) $scope.aggForm.agg.$setValidity('bucket', canUseAggregation);
+    if (canUseAggregation) {
+      if (buckets[0].type.name === 'histogram') {
+        buckets[0].params.min_doc_count = 1;
```

Comment:
min_doc_count tells elastic search what's the minimum count of documents in a bucket in order to return that bucket.

normally our setting for this is 1 ... this means elastic search wont return empty buckets (if you have date histogram split with monday, tuesday, wednesday and there is no documents for tuesday es wont return that bucket). for pipeline aggregation its required to set this to 0.

the reason why with histogram its different is that its actually inverting the value of min_doc_count in its write function ... so we need to double invert here. in the future i plan to change how histogram behaves in the first place and at that point i can take this out.